logfire/_internal/auth.py

```diff
@@ -100,25 +100,30 @@ def __str__(self) -> str:
 
 @dataclass
 class UserTokenCollection:
-    """A collection of user tokens."""
+    """A collection of user tokens, read from a user tokens file."""
 
     user_tokens: dict[str, UserToken]
     """A mapping between base URLs and user tokens."""
 
-    @classmethod
-    def empty(cls) -> Self:
-        """Create an empty user token collection."""
-        return cls(user_tokens={})
+    path: Path
+    """The path where the user tokens are stored."""
 
-    @classmethod
-    def from_file_data(cls, file_data: UserTokensFileData) -> Self:
-        return cls(
-            user_tokens={url: UserToken(base_url=url, **data) for url, data in file_data.get('tokens', {}).items()}
-        )
+    def __init__(self, path: Path) -> None:
+        self.path = path
+        data = cast(UserTokensFileData, read_toml_file(path))
+        self.user_tokens = {url: UserToken(base_url=url, **data) for url, data in data.get('tokens', {}).items()}
 
     @classmethod
-    def from_tokens_file(cls, file: Path) -> Self:
-        return cls.from_file_data(cast(UserTokensFileData, read_toml_file(file)))
+    def empty(cls, path: Path) -> Self:
+        """Create an empty user token collection.
+
+        Args:
+            path: The path where the user tokens will be stored.
+        """
+        inst = cls.__new__(cls)
+        inst.user_tokens = {}
+        inst.path = path
+        return inst
```

Comment:
Alternatively, keep the dataclass-generated `__init__()`, and re-add the classmethod constructors (but then we have a public `__init__()` that shouldn't be used because `path` and `user_tokens` shouldn't be provided separately).